r/R/dplyr-funcs-datetime.R

```diff
@@ -351,6 +351,25 @@ register_bindings_duration <- function() {
     delta <- delta$cast(int64())
     start + delta$cast(duration("s"))
   })
+    register_binding("make_difftime", function(num = NULL,
+                                             units = "secs",
+                                             ...) {
+    if (units != "secs") {
+      abort("`make_difftime()` with units other than 'secs' not supported in Arrow")
+    }
+
+    chunks <- list(...)
+
+    if (is.null(num)) {
+      duration <- duration_from_chunks(chunks)
+    } else if (length(chunks) == 0) {
+      duration <- num
+    } else {
+      abort("`make_difftime()` with both `num` and `...` not supported in Arrow")
+    }
+    duration <- build_expr("cast", duration, options = cast_options(to_type = int64()))
```

Comment:
I've seen this line in a few PRs recently — this is to get around the fact that we do not have float durations, yeah? I wonder if we should put that into a helper function. Not so much to save lines of code, but to encapsulate that we have this one workaround that we're using in all of these places. 

Then if that workaround is not necessary (or if we find a better one) we can change it in one place. We don't need to do that here, but it would be good to make a Jira for that if we do want to.

Reply:
I'd be in favour of having a separate Jira for this, so I've created one: [ARROW-16253](https://issues.apache.org/jira/browse/ARROW-16253).